contrib/aws/aws-sdk-go-v2/aws/aws.go

```diff
@@ -304,8 +312,35 @@ func stateMachineName(requestInput middleware.InitializeInput) string {
 	case *sfn.DeleteStateMachineInput:
 		stateMachineArn = *params.StateMachineArn
 	}
-	parts := strings.Split(stateMachineArn, ":")
-	return parts[len(parts)-1]
+	if stateMachineArn != "" {
+		// https://docs.aws.amazon.com/step-functions/latest/apireference/API_StartExecution.html#API_StartExecution_RequestSyntax:~:text=Required%3A%20No-,stateMachineArn,-The%20Amazon%20Resource
+		// arn:<partition>:states:<region>:<account-id>:stateMachine:<myStateMachineName>
+		// arn:<partition>:states:<region>:<account-id>:stateMachine:<myStateMachineName>:10
+		// arn:<partition>:states:<region>:<account-id>:stateMachine:<myStateMachineName:PROD>
+		// There are 3 patterns to cover and attempt to capture the `myStateMachineName`, it should always be in index 6
+		kv[tags.SFNStateMachineArn] = stateMachineArn
+		parts = strings.Split(stateMachineArn, ":")
+		stateMachineName = parts[6]
+		awsAccount = parts[4]
```

Comment:
The output is valid JSON, right? So how about just decoding the JSON instead? That'd be safer, cleaner, and more future proof.